main.py

```diff
@@ -0,0 +1,127 @@
+            for giving_partner in result:
+                print(
+                    f"Processing donee_id: {giving_partner.id}, name: {giving_partner.name}, address: {giving_partner.address}, {giving_partner.city}, {giving_partner.state}, {giving_partner.country}"  # pylint: disable=line-too-long
+                )  # log this
+                process_gp(giving_partner, session, gpl)
+    except SQLAlchemyError as e:
+        print(f"failed to create session: {e}")  # error log this
+    engine.dispose()
+
+
+def process_gp(giving_partner, session, gp_table):
+    """Module that processes each GP"""
+    autocomplete_result = autocomplete_check(giving_partner)
+    # autocomplete_result is a tuple of type (bool, place_id)
+    if autocomplete_result[0]:
+        gp_address = f"{giving_partner.address}, {giving_partner.city}, {giving_partner.state}, {giving_partner.country}"
+        gp_info = gp_table(
+            giving_partner_id=giving_partner.id,
+            phone_number=giving_partner.phone,
+            address=gp_address,
+            latitude=giving_partner.latitude,
+            longitude=giving_partner.longitude,
+            api_id=autocomplete_result[1],
+            source="Google",
+        )
+        try:
+            session.add(gp_info)
+            session.commit()
+            print(
+                f"succesfully processed {giving_partner.name}"
+            )  # log this sucessfull processing
+        except SQLAlchemyError as e:
+            print(f"sqlalchemy insertion error: {e}")  # error log this
+            raise
+        return True
+    try:
+        text_search_results = text_search(giving_partner)
+    except RuntimeError as e:
+        print(f"{e}")  # error log this
+        raise
+
+    if len(text_search_results) > 0:
+        # get the topmost result from the text search assuming it is the right GP
+        top_result = text_search_results[0]
+        valid = check_topmost(top_result, giving_partner)
+        if not valid:
+            print(
+                f"not processed as the topmost result from text search {top_result["displayName"]["text"]} does not match gp name {giving_partner.name}"  # pylint: disable=line-too-long
+            )
+            return False
+        gp_info = gp_table(
+            giving_partner_id=giving_partner.id,
+            phone_number=giving_partner.phone,
+            address=top_result["formattedAddress"],
+            latitude=top_result["location"]["latitude"],
+            longitude=top_result["location"]["longitude"],
+            api_id=top_result["id"],
+            source="Google",
+        )
+        try:
+            session.add(gp_info)
+            session.commit()
+            print(
+                f"succesfully processed {giving_partner.name}"
+            )  # log this sucessful processing
+        except SQLAlchemyError as e:
+            print(f"sqlalchemy insertion error: {e}")  # error log this
+            raise
+        return True
```

Comment:
We can't assume the shape of top_result so it is safer to add it into the try except

```suggestion
        try:
            gp_info = gp_table(
                giving_partner_id=giving_partner.id,
                phone_number=giving_partner.phone,
                address=top_result["formattedAddress"],
                latitude=top_result["location"]["latitude"],
                longitude=top_result["location"]["longitude"],
                api_id=top_result["id"],
                source="Google",
            )
        
            session.add(gp_info)
            session.commit()
            print(
                f"succesfully processed {giving_partner.name}"
            )  # log this sucessful processing
        except (SQLAlchemyError, KeyError, TypeError) as e:
            print(f"Insertion failed for {giving_partner.name}: {e}")  # log error
            raise
        return True
```